### ISSUE TITLE
Inference before finish training

### ISSUE THREAD

**Issue by sky4689524** [MIC-DKFZ/nnUNet#611]
Hello,

Firstl,y thank you for sharing this great nnUNet script on github. 

I am training nnUNet now. But I want to inference some test dataset with trained model before finish training.
Is there any way to do this?

Or do I have to wait until finish training?

**Comment by sky4689524** [comment]
I found the way how to do.
I used to train model with fold 'all' in 3d_fullres.
In this case, I need to use nnUNet_predict -i Input_folder -o output_folder -t XXX -m 3d_fullres -chk model_best -f all

**Comment by sky4689524** [comment]
But in this case, I have this warning message
"""
WARNING! Cannot run postprocessing because the postprocessing file is missing. Make sure to run consolidate_folds in the output folder of the model first!
"""

Is it because I didn't finish training?

**Comment by FabianIsensee** [comment]
If you use fold 'all' you can't use postprocesing. It doesn't do much anyways so you can ignore this.
Terminating training early is NOT recomended :-)

**Comment by QiangZhangCV** [comment]
I wonder why do inference with the last model rather than the best model. In my cases,  overfittings always occur  when training epoch number reaches 1000 and the validation dice is quite lower than validation dice with best model. Anyway to use the best model?

**Comment by FabianIsensee** [comment]
if you want valid cross-validation scores you need to use the final model for validation. You would be overfitting to the validation set if you used best instead. 
Whether you use best of final model on test set is up to you, but NEVER try both - that would be overfitting to the test set

Do not use the 'dice scores' reported during training as indications of actual validation dice. Those are completely different!

Does your green line decrease towards the end of the training?

**Comment by Jun-electrophysiology** [comment]
@FabianIsensee 
Hi Fabian,
In your above comment, you said "NEVER try both - that would be overfitting to the test set". What do you mean by "try both"? Could you please explain it detailly?

Best,
Jun

**Comment by FabianIsensee** [comment]
I am wondering a bit about myself here. Not sure what I meant to say. Anyways. You can use the best model of course, but then you need to compensate for the fact that you have been overfitting on the validation set (epoch selection).
In that case you need to either report test set scores using the selected model (in your case the best) or report the validation set scores from the final (!) model. Or both ;-)

**Comment by Jun-electrophysiology** [comment]
I got it. Thanks for your suggestion!
